projects/com.oracle.truffle.llvm.parser.bc/src/com/oracle/truffle/llvm/parser/bc/LLVMBitcodeVisitor.java

```diff
@@ -348,10 +359,74 @@ private LLVMExpressionNode getGlobalVariable(GlobalValueSymbol global) {
             final LLVMExpressionNode loadedFunction = factoryFacade.createLoad(this, functionType, functionLoadTarget);
             final LLVMExpressionNode[] argNodes = new LLVMExpressionNode[]{factoryFacade.createFrameRead(this, LLVMBaseType.ADDRESS, stackSlot)};
             final LLVMNode functionCall = factoryFacade.createFunctionCall(this, loadedFunction, argNodes, LLVMBaseType.VOID);
-            structors[i] = functionCall;
+
+            final StructureConstant structureConstant = (StructureConstant) arrayConstant.getElement(i);
+            final Integer priorityConstant = LLVMSymbolResolver.evaluateIntegerConstant(structureConstant.getElement(0));
+            assert priorityConstant != null;
+            sortedStructors.add(priorityConstant, functionCall);
+        }
+
+        return sortedStructors.getAll();
+    }
+
+    private static final class SortedLLVMNodeList {
```

Comment:
Thanks for the clarification! Would `Arrays.sort` be applicable to replace the list class? I think that maintaining our own list data structures is error-prone (maybe because I hardly ever achieve to implement a bug-free list implementation :smile:). If you think doing so is more complicated, we better keep the list implementation.